relayer/client.go

```diff
@@ -144,9 +144,9 @@ func CreateClient(ctx context.Context, src, dst *Chain, srcUpdateHeader, dstUpda
 	}
 
 	// Create the ClientState we want on 'src' tracking 'dst'
-	clientState, err := src.ChainProvider.NewClientState(dstUpdateHeader, tp, ubdPeriod, allowUpdateAfterExpiry, allowUpdateAfterMisbehaviour)
+	clientState, err := dst.ChainProvider.NewClientState(dstUpdateHeader, tp, ubdPeriod, allowUpdateAfterExpiry, allowUpdateAfterMisbehaviour)
 	if err != nil {
-		return false, fmt.Errorf("failed to create new client state for chain{%s} tracking chain{%s}: %w", src.ChainID(), dst.ChainID(), err)
+		return false, fmt.Errorf("failed to create new client state for chain{%s}: %w", dst.ChainID(), err)
```

Comment:
style nit:

No need for a change now and I've seen this in several of our codebases, not just here. 

I usually omit "failed to", "error", etc. as a prefix in an error messages. 

Othwerwise you get wrapped errors like:

```
failed to create new a: error fetching b: failed to get c
```

The above makes for a long message to read.

I'd much rather see:

```
create new a: fetching b: get c
```

Also, it's a non-nil `error` so obviously something failed. 😄 

Typically, a single upstream caller decorates the error message for human consumption. That caller is often a logger or, for example, in web2 an http handler that writes the error to a 500 response. 

That caller then would write:

```
Failed to do a thing: create new a: fetching b: get c
```

That's my 2 cents. No need for changes now.